tests/Id.test.ts

```diff
@@ -38,11 +38,33 @@ describe('Id', () => {
   });
   test('id can be used as Map keys', () => {
     const buffer = Buffer.from('hello world');
-    const id = Id.create(buffer);
+    const id = internalId.create(buffer);
     // The id is an object, so this is an object key
     const map = new Map();
     map.set(id, 'foo bar');
     expect(map.get(id)).toBe('foo bar');
     expect(map.get(id.toString())).toBeUndefined();
   });
+  type Opaque<K, T> = T & { __TYPE__: K };
+  type OpaqueId = Opaque<"opaque", Id>
+  test('id can be created as a opaque type directly', async () => {
+    const buffer = Buffer.from('abcefg');
+
+    const id1 = internalId.create<OpaqueId>(buffer);
```

Comment:
I thought idea was that you're going to also extend the `IdInternal` class so that you have extra methods just for `NodeId`... etc. how would that work?

Reply:
Still working that out.